### PR TITLE
Add preact test + add test for another variation of returning default function

### PR DESCRIPTION
**Pull request by tmcneal**

### CODE REVIEW

File: examples/defaultfunc2/src/Foo.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+export default function Foo (props) {
+  const {
+    onChange,
+    placeholder,
+    ...rest
+  } = props;
+
+  const [inputValue, setInputValue] = React.useState('');
+  const inputRef = React.useRef(null);
```
Comment (khsheehan) [inline examples/defaultfunc2/src/Foo.js:24]: Do we need this ref (same for the ref prop below)?

**Review by khsheehan — APPROVED** [review]
1 question but definitely not a blocker

**Comment by tmcneal** [comment]
Thanks for the review 👍

File: examples/defaultfunc2/src/Foo.js (hunk 1 above)
Comment (khsheehan) [reply examples/defaultfunc2/src/Foo.js:24]: Makes sense to me!

**Pull request merged**